rsocket-tck-drivers/src/main/java/io/rsocket/tckdrivers/test/TckTest.java

```diff
@@ -0,0 +1,149 @@
+
+import io.rsocket.RSocket;
+import io.rsocket.RSocketFactory;
+import io.rsocket.tckdrivers.client.JavaClientDriver;
+import io.rsocket.tckdrivers.common.ServerThread;
+import io.rsocket.transport.netty.client.TcpClientTransport;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.net.URI;
+import java.util.*;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class TckTest {
+
+  /*
+   * Start port. For every input test file a server instance will be launched.
+   * For every server instance currentPort is incremented by 1
+   */
+  private static int currentPort = 4567;
+
+  private static final String hostname = "localhost";
+
+  /* Run all the test from "path". Tests files are expected to have same names
+   * with a prefix "server" or "client" to indicate whether they type.
+   */
+  private static final String path = "src/test/resources/";
+
+  private static final String serverPrefix = "server";
+  private static final String clientPrefix = "client";
+
+  private String name; // Test name
+  private JavaClientDriver jd; // javaclientdriver object for running the given test
+  private List<String> test; // test instructions/commands
+
+  public TckTest(String name, JavaClientDriver jd, List<String> test) {
+    this.name = name;
+    this.jd = jd;
+    this.test = test;
+  }
+
+  /** Runs the test. */
+  @Test
+  public void TckTestRunner() {
+
+    try {
+      this.jd.runTest(this.test.subList(1, this.test.size()), this.name);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * A function that reads all the server/client test files from "path". For each server file, it
+   * starts a server. It parses each client file and create a parameterized test.
+   *
+   * @return interatable tests
+   */
+  @Parameters(name = "{index}: {0}")
+  public static Iterable<Object[]> data() {
+
+    File folder = new File(path);
+    File[] listOfFiles = folder.listFiles();
+    List<Object[]> testData = new ArrayList<Object[]>();
+
+    for (int i = 0; i < listOfFiles.length; i++) {
+      File file = listOfFiles[i];
+      if (file.isFile() && file.getName().startsWith(clientPrefix)) {
+        String serverFileName = serverPrefix + file.getName().replaceFirst(clientPrefix, "");
+
+        File f = new File(path + serverFileName);
+        if (f.exists() && !f.isDirectory()) {
+
+          //starting a server
+          ServerThread st = new ServerThread(currentPort, path + serverFileName);
+          st.start();
+          st.awaitStart();
+
+          try {
+            RSocket client = createClient(new URI("tcp://" + hostname + ":" + currentPort + "/rs"));
+            JavaClientDriver jd = new JavaClientDriver(() -> client);
+
+            BufferedReader reader = new BufferedReader(new FileReader(file));
```

Comment:
Not essential for this PR, but this method is quite big and could be improved by refactoring, e.g. extract methods, introduce a TckIndividualTest instead of String name, List<String> test